CMakeLists.txt

```diff
@@ -132,7 +150,13 @@ if(NOT UNITS_HEADER_ONLY)
         UNITS_BUILD_OBJECT_LIBRARY "Enable construction of the units object library"
         OFF "NOT CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME" OFF
     )
-
+else()
+    option(UNITS_BUILD_STATIC_LIBRARY "enable Construction of the units static library"
+           OFF
+    )
+    option(UNITS_BUILD_SHARED_LIBRARY "enable Construction of the units shared library"
+           OFF
+    )
 endif(NOT UNITS_HEADER_ONLY)
```

Comment:
The block:
```cmake
if(NOT UNITS_HEADER_ONLY)
...
endif(NOT UNITS_HEADER_ONLY)
```
looks a little hardcoded but I can't propose anything better.

The only thing I can propose is to check whether the developper have checked `ON` only one option from these three (`UNITS_BUILD_STATIC_LIBRARY , UNITS_BUILD_SHARED_LIBRARY, UNITS_HEADER_ONLY`) and if there are for example both `UNITS_BUILD_STATIC_LIBRARY =ON` and `UNITS_BUILD_SHARED_LIBRARY=ON` we could send a message (`WARNING` or even `FATAL_ERROR`) telling that only one library at a time can be built

Reply:
I added some warnings to check if more than one of UNITS_BUILD_STATIC_LIBRARY, UNITS_BUILD_SHARED_LIBRARY, and UNITS_BUILD_OBJECT_LIBRARY is set, and if you set all three you will get two warnings.